src/cephadm/cephadm

```diff
@@ -2930,6 +2990,12 @@ def command_deploy():
     else:
         logger.info('%s daemon %s ...' % ('Deploy', args.name))
 
+    if args.registry_url and args.registry_username and args.registry_password:
+        out, err, code = call([container_path, 'login',
+                                   '-u', args.registry_username,
+                                   '-p', args.registry_password,
+                                   args.registry_url])
```

Comment:
I think we do not need this modification. if we are using a custom registry with/without authentication we will need to use this information every time we need to deploy a new daemon (query an image). I mean, it is a check that we always will need to do , check if there is a custom registry/authentication and use it.

Reply:
I do really like the idea of just having a login in _create_daemon rather than doing the login directly in the deploy command. However, I do agree with @jmolmo that there isn't really any need for the extra check since the logins are quick. I've currently gone with a sort of middle ground where I changed back command_deploy and just do the login in the necessary places. Also, I now have it setup so it will call the login on all hosts when running registry-login in the mgr so the credentials should be distributed on the spot which also has the added bonus of allowing us to check the credentials actually work before storing them for later use.

Reply:
I am now having second thoughts about having the registry-login command login all hosts at once. I don't want storing the credentials to fail if a single host of many can't store the info. I think I will look into this type of change where the hosts are given credentials as part of the serve loop.